# This is a rebase for the 5.10.129 release.
# Original commit:  cd58ed7eb9645b2e54136cf52be6279dbdebe936
# URL:  https://github.com/google/bbr/commit/cd58ed7eb9645b2e54136cf52be6279dbdebe936

net/ipv4/tcp_rate.c:
// SPDX-License-Identifier: GPL-2.0-only

--- a/net/ipv4/tcp_rate.c.orig2	2022-07-12 16:27:51.617025080 -0700
+++ b/net/ipv4/tcp_rate.c	2022-07-12 16:28:29.654510464 -0700
@@ -105,6 +105,7 @@ void tcp_rate_skb_delivered(struct sock
 	if (!rs->prior_delivered ||
 	    tcp_skb_sent_after(tx_tstamp, tp->first_tx_mstamp,
 			       scb->end_seq, rs->last_end_seq)) {
+		rs->prior_lost	     = scb->tx.lost;
 		rs->prior_delivered_ce  = scb->tx.delivered_ce;
 		rs->prior_delivered  = scb->tx.delivered;
 		rs->prior_mstamp     = scb->tx.delivered_mstamp;
